src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,190 @@
+            try {
+                addToDo(line);
+            } catch (DukeException e) {
+                System.out.println(HORIZONTAL + "☹ OOPS!!! The description of a todo cannot be empty. \n" +
+                        HORIZONTAL);
+            }
+        } else if (line.contains(DEADLINE)) {
+            try {
+                addDeadline(line);
+            } catch (DukeException e) {
+                System.out.println(HORIZONTAL + "☹ OOPS!!! A component of the deadline task is empty. \n" +
+                        HORIZONTAL);
+            }
+        } else if (line.contains(EVENT)) {
+            try {
+                addEvent(line);
+            } catch (DukeException e) {
+                System.out.println(HORIZONTAL + "☹ OOPS!!! A component of the event task is empty. \n" +
+                        HORIZONTAL);
+            }
+        } else {
+            System.out.println(HORIZONTAL + "☹ OOPS!!! I'm sorry, but I don't know what that means :-( \n" + HORIZONTAL);
+        }
+    }
+
+    public static void listTasks() {
+        System.out.print(HORIZONTAL + "Here are the tasks in your list: \n");
+        for (Task task : tasks) {
+            System.out.println((tasks.indexOf(task) + 1) + "." + task);
+        }
+        System.out.println(HORIZONTAL);
+    }
+
+    public static void markDone(int taskDone) {
+        tasks.get(taskDone - 1).markAsDone();
+        System.out.println(HORIZONTAL + "Nice! I've marked this task as done: \n  "
+                + tasks.get(taskDone - 1) + "\n" + HORIZONTAL);
+    }
+
+    public static void delete(int taskDeleted) {
+        System.out.println(HORIZONTAL + "Noted. I've removed this task: \n  " + tasks.get(taskDeleted - 1));
+        tasks.remove(taskDeleted - 1);
+        System.out.println("Now you have " + tasks.size() + " task"
+                + (tasks.size() == 1? " " : "s ") + "in the list. \n" + HORIZONTAL);
+    }
+
+    public static void readCommand(String line) {
+        if (line.equals("list")) {
+            listTasks();
+        } else if (line.contains("done")) {
+            int taskDone = Integer.parseInt(line.substring(INDEX_OF_TASK_DONE).trim());
+            markDone(taskDone);
+        } else if (line.contains("delete")) {
+            int taskDeleted = Integer.parseInt(line.substring(INDEX_OF_TASK_DELETED).trim());
+            delete(taskDeleted);
+        } else {
+            addTask(line);
+        }
+    }
+
+    public static String readTasks() {
+        StringBuilder listOfTasks = new StringBuilder();
+        for (Task task : tasks) {
+            listOfTasks.append(task.toText()).append("\n");
+        }
+        return listOfTasks.toString();
+    }
+
+    public static void main(String[] args) {
+        sayHello();
+        String line;
+        Scanner in = new Scanner(System.in);
+        line = in.nextLine();
+        String dataFile = "data/duke.txt";
+        while (!line.equals("bye")) {
+            readCommand(line);
+            try {
```

Comment:
Instead of 2 try catch blocks, you could wrap your entire code in this main method in your try-catch block.
ie. try {
    sayHello();
    ...
    deleteFile(dataFile)
} catch (IOException e) {
    ....
}